tests/scf-upcast-custom-basis/input.dat

```diff
@@ -0,0 +1,30 @@
+molecule {
+  O 
+  H 1 0.96
+  H 1 0.96 2 104.5
+}
+
+basis low {
+    assign sto-3g
+}
+basis high {
+    assign cc-pvdz
+}
+set scf d_convergence 6
+
+
+core.IO.set_default_namespace("low")
+set basis low
+energy('hf')
+
+compare_values(-74.9634059, get_variable('SCF TOTAL ENERGY'), 6, 'SCF energy')  #TEST
+
+
+core.IO.set_default_namespace('high')
+core.IO.change_file_namespace(180, 'low', 'high')
+set guess read
+set basis high
+energy('hf')
+
+
+compare_values(-76.02663273485877, get_variable('SCF TOTAL ENERGY'), 6, 'SCF energy')  #TEST
```

Comment:
One thing that could useful here is:

```
compare_values(True, get_variable('SCF ITERATIONS') < *something*, 6, 'SCF Iterations')  #TEST
```

Just to double check the casting is working properly. The tolerance can be found with and without casting.